### DISCUSSION TITLE
Does seaweed have an access control (ACL) concept?

### DISCUSSION THREAD

**Discussion by aimg2** [seaweedfs/seaweedfs#5986] [General]
Is there documentation for access control?
I want to be able to mount a directory using "weed mount" on a VM.
The VM should be restricted to its folder and if "weed mount" is remounted to another folder it should fail.
Is this supported? Can we build this by making a reverse proxy that applies the ACL?

**Comment by Material-Scientist** [comment]
It's not obvious from the documentation, but the best way to manage access rights for SeaweedFS (IMO) is to use S3/iam.

Instructions for creating accounts and defining ACLs can be found [here](https://github.com/seaweedfs/seaweedfs/wiki/AWS-IAM-CLI).

So, you would grant your VM access to a particular bucket or subdirectory on a bucket, and it will then only have access to that.

**Reply by pschoen-itsc** [reply]
I just tested this with subdirectories and it does not seem to work. If I try to create a policy for a subdirectory (even in an already existing bucket), I get `I1223 16:22:37.665459 iamapi_management_handlers.go:337 not match bucket: [kubeflow alice *]` and no policies will be applied to `s3.configure`

**Reply by Material-Scientist** [reply]
![image](https://github.com/user-attachments/assets/ad15c571-e8aa-4a97-bf4d-d64769ca0821)

What you're referring to is due to the code expecting 2 parts when the path is split by a '/'. It's actually an unreported bug that even the docs get wrong:
![image](https://github.com/user-attachments/assets/7d0a51af-ed63-4573-94d7-dad75d66ba87)

So:
`arn:aws:s3:::bucket-name/path/*` or `arn:aws:s3:::bucket-name/*`
Instead of:
`arn:aws:s3:::bucket-name*` or `arn:aws:s3:::bucket-name`

cc:
@chrislusf

**Comment by aimg2** [comment]
This is an interesting method.
Does this method work for filer fuse integration. For example via weed mount?
